diff --git a/node_modules/react-native-toast-message/lib/src/ToastUI.js b/node_modules/react-native-toast-message/lib/src/ToastUI.js
index bb5ce5c..2720e98 100644
--- a/node_modules/react-native-toast-message/lib/src/ToastUI.js
+++ b/node_modules/react-native-toast-message/lib/src/ToastUI.js
@@ -34,6 +34,7 @@ function renderComponent({ data, options, config, isVisible, show, hide }) {
 export function ToastUI(props) {
     const { isVisible, options, hide } = props;
     const { position, topOffset, bottomOffset, keyboardOffset } = options;
+    if (!isVisible) return (<></>);
     return (<AnimatedContainer isVisible={isVisible} position={position} topOffset={topOffset} bottomOffset={bottomOffset} keyboardOffset={keyboardOffset} onHide={hide}>
       {renderComponent(props)}
     </AnimatedContainer>);
